### ISSUE TITLE
sonar settings not taken into consideration by the task

### ISSUE THREAD

**Issue by baywet** [Microsoft/vso-agent-tasks#744]
Hi,
I set the property
sonar.exclusions=**/*.min.js in /opt/sonar/conf/sonar.properties
But the sonar Task was still analyzing min.js files
Then I set it at the sonar project level, same issue. (using the exclusions UI)
Then I set /d:sonar.exclusions=**/*.min.js as Additional settings of the Task and it started properly.
As I workaround my issue this is not critical to me but I think it should work properly.
Regards

**Comment by dbolkensteyn** [comment]
@baywet First of all, please use StackOverflow with the `sonarqube` and `msbuild` tags in the future for this kind of questions.

You probably are referring to the SonarQube Scanner for MSBuild Tasks (`SonarQube for MSBuild - Begin Analysis` and `SonarQube for MSBuild - End Analysis`)

The `sonar.properties` file that you mention is only meant to store the configuration settings of the SonarQube server itself. You cannot put any project settings in it. Instead, you can set them through the UI, or programmatically using SonarQube web services calls.

The SonarQube Scanner (previously named `sonar-runner`) has a `sonar-project.properties` file in which you can store project settings. However, to analyze MSBuild projects, you (and the tasks) are using the SonarQube Scanner for MSBuild (previously named `MSBuild SonarQube Runner`) which itself does not understand the `sonar-project.properties` file.

So, to sum up, you can set the setting at these levels (in increasing order of priority):
1. Globally on the SonarQube server, either through the UI or via web services
2. Just on one project on the SonarQube server ,either through the UI or via web services
3. In the `SonarQube.Settings.xml` file containing the global settings of the SonarQube Scanner for MSBuild (you can specify a custom one in the Task)
4. Through the additional settings, which will get passed to the command line arguments of the SonarQube Scanner for MSBuild

**Comment by dbolkensteyn** [comment]
Can this issue be closed? cc @bgavrilMS

**Comment by baywet** [comment]
Thanks @dbolkensteyn and sorry for the late answer, I was on holidays.
Ok I'll stop using that file! And yes the issue can be closed (already done)
